test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1298,6 +1300,45 @@
             (Thread/sleep 100))
           (is (= {:received-commands 1 :executed-commands 1} (stats))))))))
 
+(deftest date-round-trip
+  (svc-utils/with-puppetdb-instance
+    (let [pdb (get-service svc-utils/*server* :PuppetDBServer)
+          dispatcher (get-service svc-utils/*server* :PuppetDBCommandDispatcher)
+          shared-globals (partial cli-svc/shared-globals pdb)
+          enqueue-command (partial enqueue-command dispatcher)
+          globals (shared-globals)
+          {{:keys [connection endpoint]} :command-mq} globals
+          deactivate-ms 14250331086887
+          ;; The problem only occurred if you passed a Date to
+          ;; enqueue, a DateTime wasn't a problem.
+          input-stamp (java.util.Date. deactivate-ms)
+          expected-stamp (DateTime. deactivate-ms DateTimeZone/UTC)]
+      (enqueue-command connection endpoint
+                       (command-names :deactivate-node) 3
+                       {:certname "foo.local" :producer_timestamp input-stamp})
+      (svc-utils/wait-for-server-processing svc-utils/*server*)
+      ;; While we're here, check the value in the database too...
+      (is (= expected-stamp
+             (jdbc/with-transacted-connection
+               (:scf-read-db globals)
+               :repeatable-read
+               (from-sql-date (scf-store/node-deactivated-time "foo.local")))))
+      (is (= expected-stamp
+             (-> (client/get (str (utils/base-url->str svc-utils/*base-url*)
+                                  "/nodes")
+                             {:accept :json
+                              :throw-exceptions true
+                              :throw-entire-message true
+                              :query-params {"query"
+                                             (json/generate-string
+                                              ["or" ["=" ["node" "active"] true]
+                                               ["=" ["node" "active"] false]])}})
+                 :body
+                 json/parse-string
+                 first
+                 (get "deactivated")
+                 (pt/from-string)))))))
+
```

Comment:
This is fairly heavyweight - In the longer run, I wonder it might make sense to have a more general roundtrip test (if we don't already) that tests more than just a producer_timestamp against the instance.  Hmm, or would an "echo" service be interesting for general testing of the full encoding/decoding path?